lightning/src/chain/channelmonitor.rs

```diff
@@ -2041,8 +2052,23 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 		// Prune HTLCs from the previous counterparty commitment tx so we don't generate failure/fulfill
 		// events for now-revoked/fulfilled HTLCs.
 		if let Some(txid) = self.prev_counterparty_commitment_txid.take() {
-			for &mut (_, ref mut source) in self.counterparty_claimable_outpoints.get_mut(&txid).unwrap() {
-				*source = None;
+			let cur_claimables = self.counterparty_claimable_outpoints.get(
+				&self.current_counterparty_commitment_txid.unwrap()).unwrap();
+			for (_, ref source_opt) in self.counterparty_claimable_outpoints.get(&txid).unwrap() {
+				if let Some(source) = source_opt {
+					let mut also_in_cur_commitment = false;
+					for (_, ref cur_source_opt) in cur_claimables {
+						if cur_source_opt == source_opt {
+							also_in_cur_commitment = true;
```

Comment:
But if for some reason you don't want use `any` as in the suggestion above, you can add a break here to avoid unnecessary looping the remaining elements